tools/c7n_sphinxext/c7n_sphinxext/docgen.py

```diff
@@ -176,16 +159,40 @@ def _main(provider, output_dir, group_by):
 
     files = []
 
-    # Write out resources by grouped page
-    for key, group in itertools.groupby(
-            sorted(provider_class.resources.values(), key=group_by), key=group_by):
-        rpath = os.path.join(output_dir, "%s.rst" % key)
+    groups = {}
+
+    for r in provider_class.resources.values():
+        group = group_by(r)
+        if type(group) != list:
+            group = [group]
+        for g in group:
+            if g not in groups:
+                groups[g] = list()
+            groups[g].append(r)
+
+    # Create individual resources pages
+    for r in provider_class.resources.values():
```

Comment:
why the switch from the resource directive to writing out individual pages, this feels a little odd, if their just being included but still part of the rendered tree output, their being processed twice since their not removed.

Reply:
This is to avoid duplication. Some items can belong to multiple groups (AKS belongs to Compute & Containers).

Currently, I created single page for each resource (I think this is better approach because some resources might have a lot of filters\actions). So, we render it once and then we create pages for each group where we have only links to the resource pages.

Reply:
jinja rendering is fast, sphinx not so much, so rendering twice is fine, but have a few dozen extra files in the sphinx rst source dir isn't great.

Reply:
talked to kiril in chat, the resource pages here are actually used as the group page has change to be just links to them.